src/wire/ipv4.rs

```diff
@@ -120,6 +120,45 @@ impl Cidr {
         self.prefix_len
     }
 
+    /// Return the netmask address of this IPv4 CIDR block.
+    pub fn netmask(&self) -> Address {
+        let mut addr = [0u8; 4];
+        if self.prefix_len >= 8 {
+            for idx in 0usize..((self.prefix_len - self.prefix_len % 8) / 8) as usize {
+                addr[idx] = 255u8;
+            }
+        }
+        Address(addr)
+    }
+
+    /// Return the broadcast address of this IPv4 CIDR block.
+    pub fn broadcast(&self) -> Address {
+        let mut addr = [0u8; 4];
+        let mask_addr = self.netmask();
+        for idx in 0usize..4 {
+            if mask_addr.0[idx] == 255 {
+                addr[idx] = self.address.0[idx];
+            } else {
+                addr[idx] = 255u8;
+            }
+        }
+        Address(addr)
+    }
+
+    /// Return the network CIDR of this IPv4 CIDR block.
+    pub fn network(&self) -> Cidr {
```

Comment:
This method doesn't make any sense to me. What is "network CIDR"? `self` is already a CIDR.

Reply:
CIDR not eq network.
```
cirdr = 192.168.1.100/24
network = 192.168.1.0/24
```

Reply:
OK, the documentation is confusing. How about:

    /// Return the network block of this IPv4 CIDR.

Reply:
ok